pkg/controller/plan/adapter/ocp/builder.go

```diff
@@ -324,6 +339,148 @@ func (r *Builder) mapDisks(sourceVm *cnv.VirtualMachine, targetVmSpec *cnv.Virtu
 	}
 }
 
+type envMap struct {
+	envResource interface{}
+	volName     string
+}
+
+func (r *Builder) createEnvMaps(sourceVm *cnv.VirtualMachine, vmRef ref.Ref) (map[string]*envMap, map[string]*envMap) {
+	configMaps := make(map[string]*envMap)
+	secrets := make(map[string]*envMap)
+
+	for _, envVol := range sourceVm.Spec.Template.Spec.Volumes {
+		switch {
+		case envVol.ConfigMap != nil:
+			configMap := &core.ConfigMap{}
+			err := r.sourceClient.Get(context.Background(), client.ObjectKey{Namespace: vmRef.Namespace, Name: envVol.ConfigMap.Name}, configMap)
+			if err != nil {
+				r.Log.Error(err, "Failed to get ConfigMap", "namespace", vmRef.Namespace, "name", envVol.ConfigMap.Name)
+				continue
+			}
+			configMaps[envVol.ConfigMap.Name] = &envMap{
+				envResource: configMap,
+				volName:     envVol.Name,
+			}
+
+		case envVol.Secret != nil:
+			secret := &core.Secret{}
+			err := r.sourceClient.Get(context.Background(), client.ObjectKey{Namespace: vmRef.Namespace, Name: envVol.Secret.SecretName}, secret)
+			if err != nil {
+				r.Log.Error(err, "Failed to get Secret", "namespace", vmRef.Namespace, "name", envVol.Secret.SecretName)
+				continue
+			}
+			secrets[envVol.Secret.SecretName] = &envMap{
+				envResource: secret,
+				volName:     envVol.Name,
+			}
+		}
+	}
+
+	return configMaps, secrets
+}
+
+func (r *Builder) mapConfigMapsToTarget(targetVmSpec *cnv.VirtualMachineSpec, configMaps map[string]*envMap, diskMap map[string]*cnv.Disk) {
+	for _, configMap := range configMaps {
+		// Create configmap on destination cluster
+		sourceConfigMap := configMap.envResource.(*core.ConfigMap)
+		targetConfigMap := &core.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        sourceConfigMap.Name,
+				Namespace:   r.Plan.Spec.TargetNamespace,
+				Labels:      sourceConfigMap.Labels,
+				Annotations: sourceConfigMap.Annotations,
+			},
+			Data: sourceConfigMap.Data,
+		}
+		err := r.Destination.Client.Create(context.Background(), targetConfigMap)
```

Comment:
i think all the client create should be done in another place (like kubevirt.go but with OCP maybe we can use destination client for that). can be done in another PR, if there is an agreement..

Reply:
even with other source provider types, the client that is used to post entities on the target cluster should be Destination.Client - when we migrate to the local cluster, it's the same client we use for posting the management entities on the local cluster

I agree it would be nicer to generalize the process in a way that builders return back configmaps and secrets from the source but as it looks specific to OCP, it doesn't seem to be cost-effective.. do you see data that should be stored in secrets or configmaps from other sources? (leave TPM aside as we don't get it from RHV and from vSphere it would probably not be useful)

Reply:
nope i don't see that data is needed from other sources.